shard.go

```diff
@@ -234,7 +234,9 @@ func (s *Shard) close() error {
 	if s.store != nil {
 		_ = s.store.Close()
 	}
-	s.stats.Inc("close")
```

Comment:
Yes, this is why we need to continue to rework this code. Here:

https://github.com/influxdb/influxdb/blob/master/server.go#L318

we add shards, but never open them. In this function nothing prevents `close` being called on a shard object that was never opened.